### PR TITLE
Refactor base components with <script setup>

### PR DESCRIPTION
**Pull request by samuveth**

### CODE REVIEW

File: src/components/TextareaAutosize.vue [hunk 1]
```diff
@@ -1,121 +1,122 @@
...
+watch(modelValue, v => {
+  val.value = v;
+});
+
+watch(val, v => {
+  nextTick(resize);
+  emit('update:modelValue', v);
+});
+
+watch(minHeight, () => {
```
Comment (bonustrack) [inline src/components/TextareaAutosize.vue:107]: You could watch multiple source here like this 
```suggestion
watch([minHeight, maxHeight], () => {
```
See https://v3.vuejs.org/guide/reactivity-computed-watchers.html#watching-multiple-sources

**Review by bonustrack — COMMENTED** [review]
Seem to work well too

**Comment by samuveth** [comment]
> Seem to work well too

Yes, good to know

**Comment by ChaituVR** [comment]
What's the difference between this PR and #485 ??

**Comment by samuveth** [comment]
> What's the difference between this PR and #485 ??

https://github.com/snapshot-labs/snapshot/pull/485 is just a continuation. I split things up to make testing easier.

**Comment by samuveth** [comment]
@bonustrack @ChaituVR please approve when reviewed

**Review by ChaituVR — APPROVED** [review]
LGTM (checked in #485)

**Pull request merged**